include/up-cpp/datamodel/builder/Uuid.h

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-FileType: SOURCE
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_DATAMODEL_BUILDER_UUID_H
+#define UP_CPP_DATAMODEL_BUILDER_UUID_H
+
+#include <uprotocol/v1/uuid.pb.h>
+
+#include <chrono>
+#include <cstdint>
+#include <functional>
+#include <memory>
+
+namespace uprotocol::datamodel::builder {
+
+/// @brief Builder class designed to build UUID v8 objects for uProtocol.
+///
+/// The UUID is based off the draft-ietf-uuidrev-rfc4122bis and UUIDv7 with
```

Comment:
I think it's based off UUIDv8?

Reply:
I kept Michael's documentation here, so I don't really know. If it is inaccurate, I can update the comment. Do you have a reference for the UUIDv8 RFC?

Reply:
The [UUID uProtocol spec](https://github.com/eclipse-uprotocol/up-spec/blob/main/basics/uuid.adoc) has some references that could replace these.

Reply:
@gregmedd we can ditch the other types and stick with only v8, don't spend any cycles on supporting other formats